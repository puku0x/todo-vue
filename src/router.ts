import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router';

const routes: Array<RouteRecordRaw> = [
  { path: '/', redirect: '/todos' },
  {
    path: '/todos',
    component: () => import('./pages/todo/todo-list').then(m => m.TodoListPage)
  },
  {
    path: '/todos/new',
    component: () =>
      import('./pages/todo/todo-create').then(m => m.TodoCreatePage)
  },
  {
    path: '/todos/:id',
    component: () =>
      import('./pages/todo/todo-detail').then(m => m.TodoDetailPage)
  },
  {
    path: '/todos/:id/edit',
    component: () => import('./pages/todo/todo-edit').then(m => m.TodoEditPage)
  }
];

export const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes
});
