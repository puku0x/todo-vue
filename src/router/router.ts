import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router';

const routes: Array<RouteRecordRaw> = [
  { path: '/', redirect: '/todos' },
  {
    path: '/todos',
    component: () => import('../pages/todo').then(m => m.TodoListPage)
  }
];

export const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes
});
