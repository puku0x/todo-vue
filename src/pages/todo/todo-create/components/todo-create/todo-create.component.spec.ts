import { mount } from '@vue/test-utils';
import { createRouter, createMemoryHistory } from 'vue-router';

import { routes } from '@/router';

import TodoCreate from './todo-create.component.vue';

describe('TodoCreate', () => {
  it('render', async () => {
    const router = createRouter({
      history: createMemoryHistory(),
      routes,
    });
    router.push('/todos/new');
    await router.isReady();
    const wrapper = mount(TodoCreate, {
      global: {
        plugins: [router],
      },
      props: {
        isFetching: false,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
