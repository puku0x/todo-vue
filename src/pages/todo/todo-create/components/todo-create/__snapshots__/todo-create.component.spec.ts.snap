// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoCreate render 1`] = `
<a href="/todos" class="">back to list</a>
<h2>todo-create</h2>
<form novalidate="">
  <p><button type="submit" disabled=""> Save </button></p>
  <table>
    <tbody>
      <tr>
        <td>title</td>
        <td><input type="text" name="title"></td>
      </tr>
    </tbody>
  </table>
</form>
`;
